
// THIS FILE IS AUTOMATICALLY GENERATED FROM ndap-spec.org using org-babel-mode
// Do not edit it by hand!  Edit the original and re-generate
// Instructions are found in ndap-spec.org!
//
// Contact Eric Anderson <andersoe@cs.cmu.edu> with questions.

message NetDescriptor {
        optional bytes GUID = 1;
        optional LayerTwoIdentifier l2_id = 2;
        required AuthCapStruct auth_cap = 3;
}

message LayerTwoIdentifier {
        
        // Identifies the type of Layer 2 Network referred to by this Descriptor        
        enum LayerTwoType {
                ETHERNET = 0;
                WIFI = 1;
                DSRC = 2;
        }
        required LayerTwoType l2_type =1;

        // Meaning of l2id is left up to L2-specific implementation.
        optional bytes l2_net_id = 2;
}

message ACNode {
        optional JoinAuthNode ja = 1;
        optional XIPNode xip = 2;
        optional uint32 id = 3;
        // TO BE EXTENDED
}

message JoinAuthNode {
        required uint32 version =1;
        required uint32 aaa_provider=2;
        optional uint32 context=3;              // Who knows if this will be used
}

message XIPNode {
        required bool is_global = 1;
        optional bytes NetworkId = 2;
}

message AuthCapStruct {
        repeated ACNode nodes = 1;
        repeated ACEdge edges = 2;
}

message ACEdge {
        // Indexed base 1
        required uint32 from_node =1;
        required uint32 to_node = 2;
}
